clustered/server/src/main/java/org/ehcache/clustered/server/state/MessageTracker.java

```diff
@@ -80,6 +85,17 @@ public boolean isEmpty() {
    * Remove the contiguous seen msgIds from the nonContiguousMsgIds and update highestContiguousMsgId
    */
   private void reconcile() {
+
+    // This happens when a passive is started after Active has moved on and
+    // passive starts to see msgIDs starting from a number > 0.
+    if (highestContiguousMsgId == -1L && nonContiguousMsgIds.size() > 100) {
```

Comment:
I do not like that magic number of `100`.

Also when looking more deeply at the class, we should replace the `Set` by a `ConcurrentSkipListSet` which would allow us to simply iterate over it instead of having the `contains` and separated `remove` calls.